lib/logstash/outputs/dynatrace.rb

```diff
@@ -103,6 +106,40 @@ def run
         end
       end
 
+      class BatchSerializer
+        def initialize(max_batch_size)
+          @max_batch_size = max_batch_size
+          @batch_events_size = 0
+          @serialized_events = []
+        end
+
+        def offer(event)
+          serialized_event = LogStash::Json.dump(event.to_hash)
+          if batch_size + serialized_event.length + (@serialized_events.length.positive? ? 1 : 0) > @max_batch_size
+            return false
+          end
+
+          @serialized_events.push(serialized_event)
+          @batch_events_size += serialized_event.length
+          true
+        end
+
+        def batch_size
```

Comment:
This sounds like the number of items in the batch, maybe call it `batch_wire_size` or `batch_size_serialized` or something like that.